.github/workflows/checks.yml

```diff
@@ -21,5 +21,9 @@ jobs:
           extra_nix_config: |
             experimental-features = nix-command flakes
             access-tokens = github.com:${{ secrets.NIX_GITHUB_TOKEN }}
-      - name: build all flake templates
-        run: ./checks
+      - name: validate root devShell
+        run: nix develop -c echo "PASSED"
+      - name: validate formatting of all files
+        run: nix develop -c treefmt --fail-on-change
+      - name: build all flake templates on macOS and linux
+        run: nix develop -c checks
```

Comment:
Can we make the checks script a flake check instead? That way, instead of `nix develop -c checks`, just run `nix flake check`.